### ISSUE TITLE
Readme (npm) install instructions fail

### ISSUE THREAD

**Issue by clounie** [GoogleCloudPlatform/gcloud-node#1466]
#### Steps to reproduce

  1. `npm install google-cloud`
  2. It fails, because https://www.npmjs.com/package/google-cloud doesn't exist. The [gcloud](https://www.npmjs.com/package/gcloud) package appears to have the last release in npm though.

So, either readme or npm package name needs updated.

**Comment by stephenplusplus** [comment]
Yeah, our repo just went through a major restructure. We're going to be splitting into multiple modules, so that you can `npm install @google-cloud/storage` for just the Storage module for example, or `npm install google-cloud` and get the whole kit of services. For now, the readme won't be relevant in that regard.

You can browse the state of the repo at the latest npm release of gcloud-node (0.37.0) [here](https://github.com/GoogleCloudPlatform/gcloud-node/tree/v0.37.0) and check out the docs site that is accurate for that release [here](https://googlecloudplatform.github.io/gcloud-node).

**Comment by jasonpolites** [comment]
IMO we should update the README to reflect what works.  Currently anyone who visits the GCP docs for Node.js (https://cloud.google.com/nodejs/apis) will end up in a broken state because npm install will fail.

Also the docs at https://cloud.google.com/nodejs/apis which suggest `require('google-cloud')` do align with the instruction to `npm install --save gcloud`

**Comment by jgeewax** [comment]
Yea our readme can't lie... @stephenplusplus We need to fix this.

**Comment by stephenplusplus** [comment]
https://cloud.google.com/nodejs/apis -- the link there goes to the docs site, https://googlecloudplatform.github.io/gcloud-node/#/, which is correct. Did I miss the link to the GH repo?

> Also the docs at https://cloud.google.com/nodejs/apis which suggest `require('google-cloud')` do align with the instruction to `npm install --save gcloud`

I didn't see anything about require('google-cloud') on that site.

**Comment by stephenplusplus** [comment]
I'll put the old readme back. Still curious about the questions above, though. Links would be appreciated!

**Comment by jasonpolites** [comment]
This is what I see at https://googlecloudplatform.github.io/gcloud-node/#/

<img width="1671" alt="screen shot 2016-08-08 at 12 06 31 pm" src="https://cloud.githubusercontent.com/assets/764528/17492425/a3794a20-5d60-11e6-8a6d-7720c23696c7.png">

But also clicking through to the README gives conflicting/confusing information that doesn't seem to work.

**Comment by stephenplusplus** [comment]
Oh, the code sample there is old, I see. I'll update that now as well. Thanks for pointing that out.

**Comment by jgeewax** [comment]
TL;DR: README is an anomaly. It current is "master". It shouldn't be master, it should be "latest" (currently v0.37). Right?

**Comment by stephenplusplus** [comment]
We always work on master and put our releases under git tags. Should that change?

**Comment by jgeewax** [comment]
README is a special case... as we're seeing here... What are our options? The current one (where we just have README on master, and our docs default to 0.37, and npm install gcloud defaults to 0.7) isn't working... :(

**Comment by stephenplusplus** [comment]
What I'm used to is going to GH and getting the latest state of the codebase, so `master` is what I expect.  I know things will be different from the latest release.

On the flip side, when I go to a "presentation" site, i.e. our docs site, I expect it to be talking about the latest release.

I do have two lessons learned through all of this, though:

  1. Leaving a note in the readme that something is changing is just plain nice, so don't forget to do that (oops)
  2. Know your own docs site-- I didn't realize it said `var gcloud = require('google-cloud')`

If we used the master branch as "the latest release" branch, we would just have a "next" branch, that we would treat the same as we treat master now. That solves this readme case, but might be unexpected for users coming to contribute-- they might do their work on the wrong branch, not realizing there is something else... or in general, not discover the *actual* state of the code base, and react accordingly (filing bugs, PRs).

**Comment by jasonpolites** [comment]
I think you want to allow for the case of customers arriving directly at GH without any other context.  It might be a link from a SO post, or just a direct result from a Google search (search for "Google Cloud Node" and see what you get, second result for me is GH).

In either case, following the README should "work", regardless of whether it's a newer version than the current release.   Generally though I think your intuition is appropriate.  Master is "latest", everything else is a tag.  I would suggest that it's "nice" to have a clear indication in the master README that it's **not** the same as the latest public release version though (e.g. a link to the latest tagged version)
